tardis/energy_input/tests/test_gamma_ray_transport.py

```diff
@@ -0,0 +1,195 @@
+import os
+import pytest
+import numpy as np
+import numpy.testing as npt
+from pathlib import Path
+import radioactivedecay as rd
+
+import tardis
+from tardis.model import SimulationState
+from tardis.io.configuration import config_reader
+from tardis.io.util import yaml_load_file, YAMLLoader
+from tardis.energy_input.gamma_ray_transport import (
+    calculate_shell_masses,
+    create_isotope_dicts,
+    get_all_isotopes,
+    create_inventories_dict,
+    calculate_total_decays,
+)
+import astropy.units as u
+import astropy.constants as c
+
+
+@pytest.fixture(scope="module")
+def gamma_ray_config(example_configuration_dir: Path):
+    """
+    Parameters
+    ----------
+    example_configuration_dir: Path to the configuration directory.
+
+    Returns
+    -------
+    Tardis configuration
+    """
+    yml_path = (
+        example_configuration_dir
+        / "tardis_configv1_density_exponential_nebular.yml"
+    )
+
+    return config_reader.Configuration.from_yaml(yml_path)
+
+
+@pytest.fixture(scope="module")
+def simulation_setup(gamma_ray_config):
+    """
+    Parameters
+    ----------
+    gamma_ray_config:
+
+    Returns
+    -------
+
+    """
+    gamma_ray_config.model.structure.velocity.start = 1.0 * u.km / u.s
+    gamma_ray_config.model.structure.density.rho_0 = 5.0e2 * u.g / (u.cm**3)
+    gamma_ray_config.supernova.time_explosion = 1.0 * (u.d)
+    model = SimulationState.from_config(gamma_ray_config)
+    return model
+
+
+def test_calculate_shell_masses(simulation_setup):
+    """
+    Function to test calculation of shell masses.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    """
+    model = simulation_setup
+    volume = model.volume.to("cm^3")
+    density = model.density.to("g/cm^3")
+    desired = (volume * density).to(u.g).value
+
+    shell_masses = calculate_shell_masses(model).value
+    npt.assert_almost_equal(shell_masses, desired)
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56"])
```

Comment:
Add one or two more nuclides.

Reply:
but it is not resolved - there is still just one?

Reply:
No. Three. Ni56, Cr48, and Fe52